python/hail/methods/impex.py

```diff
@@ -382,11 +404,49 @@ def import_locus_intervals(path, reference_genome='default', skip_invalid_interv
     :class:`.Table`
         Interval-keyed table.
     """
-    rg = reference_genome._jrep if reference_genome else None
 
-    t = Env.hail().table.Table.importIntervalList(Env.hc()._jhc, path, joption(rg),
-                                                  skip_invalid_intervals)
-    return Table(t)
+    t = import_table(path, comment="@", impute=False, no_header=True,
+                     types={'f0': tstr, 'f1': tint32, 'f2': tint32,
+                            'f3': tstr, 'f4': tstr})
+
+    if t.row.dtype == tstruct(f0=tstr):
+        if reference_genome:
+            t = t.select(interval=hl.parse_locus_interval(t['f0'], reference_genome))
+        else:
+            interval_regex = "([^:]*):(\\d+)\\-(\\d+)"
+
+            def checked_match_interval_expr(match):
+                return hl.cond(hl.len(match) == 3,
```

Comment:
or_missing

Reply:
I actually can't use `or_missing` here. `or_missing` evaluates both branches regardless of whether the predicate is true or false. `cond` does not do that.

Reply:
really? That seems like a bug...

Reply:
or_missing should just be implemented in terms of cond in Python.

Reply:
I changed it to only be implemented in Python. Removed all Scala "orMissing" functions. Not a bug -- you can't know what the type of the null value (`TTHr`) is without evaluating the first statement.